File: richest-customer-wealth/Solution1672.java
// https://leetcode.com/problems/richest-customer-wealth/

class Solution1672 {

    // T: O(n), S: O(1)
    public int maximumWealth(int[][] accounts) {
        var maxWealth = 0;
        for (var acct : accounts) {
            var acctSum = 0;
            for (var bal : acct) {
                acctSum += bal;
            }
            maxWealth = Math.max(acctSum, maxWealth);
        }
        return maxWealth;
    }

    public static void main(String[] args) {
        var sol = new Solution1672();
        var accts1 = new int[][] {
                { 1, 2, 3 },
                { 3, 2, 1 }
        };
        System.out.println(sol.maximumWealth(accts1)); // 6
    }

}

File: richest-customer-wealth/solution1672.go
// https://leetcode.com/problems/richest-customer-wealth/

package main

import (
	"fmt"
)

// T: O(n), S: O(1)
func maximumWealth(accounts [][]int) int {
	maxWealth := 0
	for _, acct := range accounts {
		acctSum := 0
		for _, bal := range acct {
			acctSum += bal
		}
		if acctSum > maxWealth {
			maxWealth = acctSum
		}
	}
	return maxWealth
}

func main() {
	accts1 := [][]int{
		{1, 2, 3},
		{3, 2, 1},
	}
	fmt.Println(maximumWealth(accts1)) // 6
}
